sdk/loadtestservice/load-testing-rest/src/beginCreateOrUpdateTestRun.ts

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+import { CancelOnProgress, OperationState, SimplePollerLike } from "@azure/core-lro";
+import { TestRunStatusPoller, PolledOperationOptions } from "./models";
+import { AzureLoadTestingClient, TestRunCreateOrUpdateParameters } from "./index.js";
+import { TestRunGet200Response } from "./responses";
+import { isUnexpected } from "./isUnexpected";
+import { sleep } from "./util/sleepLROUtility";
+
+/**
+ * Creates a poller to poll for test run status.
+ * @param client - The Load Testing client.
+ * @param options - The operation options.
+ * @returns A poller which can be called to poll until completion of the job.
+ */
+export async function beginCreateOrUpdateTestRun(
+  client: AzureLoadTestingClient,
+  testRunId: string,
+  testRunParams: TestRunCreateOrUpdateParameters,
+  polledOperationOptions: PolledOperationOptions = {}
+): Promise<TestRunStatusPoller> {
+  // Creating the test run
+  const testRunCreationResult = await client
+    .path("/test-runs/{testRunId}", testRunId)
+    .patch(testRunParams);
+
+  if (isUnexpected(testRunCreationResult)) {
+    throw testRunCreationResult.body.error;
+  }
+
+  if (testRunCreationResult.body.testRunId === undefined)
+    throw new Error("Test Run ID returned as undefined.");
+
+  type Handler = (state: OperationState<TestRunGet200Response>) => void;
+
+  const state: OperationState<TestRunGet200Response> = {
+    status: "notStarted",
+  };
+
+  const progressCallbacks = new Map<symbol, Handler>();
+  const processProgressCallbacks = async (): Promise<void> =>
+    progressCallbacks.forEach((h) => h(state));
+  let resultPromise: Promise<TestRunGet200Response> | undefined;
+  let cancelJob: (() => void) | undefined;
+  const abortController = new AbortController();
+  const currentPollIntervalInMs = polledOperationOptions.updateIntervalInMs ?? 2000;
+
+  const poller: SimplePollerLike<OperationState<TestRunGet200Response>, TestRunGet200Response> = {
+    async poll(_options?: { abortSignal?: AbortSignalLike }): Promise<void> {
+      let getTestRunResult = await client.path("/test-runs/{testRunId}", testRunId).get();
+      if (isUnexpected(getTestRunResult)) {
+        throw getTestRunResult.body.error;
+      }
```

Comment:
We shouldn't throw. We should set the status and error fields in the poller state instead.

Reply:
Done.